dask/dataframe/core.py

```diff
@@ -1178,6 +1178,64 @@ def repartition(
         elif freq is not None:
             return repartition_freq(self, freq=freq)
 
+    def shuffle(
+        self,
+        index,
+        npartitions=None,
+        max_branch=None,
+        shuffle=None,
+        shuffle_dtype=None,
+        compute=None,
+    ):
+        """ Rearrange DataFrame into new partitions by index
+
+        Uses hashing to map rows to output partitions. After this operation,
+        rows with the same index element(s) will be in the same partition.
+
+        Parameters
+        ----------
+        index : str, list of str, or _Frame
```

Comment:
```suggestion
        index : str, list of str, or Series, Index, or DataFrame
```

`_Frame` isn't public so best to keep it out of docstrings.